### PR TITLE
Use {base_url}/api for checking hub version

### PR DESCRIPTION
**Pull request by jabbera**
Disclosure: I've not actually run this.

Fixes: https://github.com/jupyterhub/jupyterhub/issues/4566

### CODE REVIEW

**Comment by minrk** [comment]
Rather than following redirects, perhaps we should fix the URL to be `{base_url}/api/`?

**Comment by jabbera** [comment]
@minrk all set.

File: jupyterhub/objects.py [hunk 1]
```diff
@@ -174,7 +174,7 @@ def wait_up(self, timeout=10, http=False, ssl_context=None):
             )
 
             return wait_for_http_server(
-                self.url, timeout=timeout, ssl_context=ssl_context
+                urljoin(self.url, "api"), timeout=timeout, ssl_context=ssl_context
```
Comment (minrk) [inline jupyterhub/objects.py:177]: This `/api/` bit is specific to waiting for user servers, which the Server is for any http server, so I think either `wait_up` needs to add a path argument or the Server on which `wait_up` is called needs to be constructed via `Server.from_url(urljoin(server.url, "api")` to add it to `self.url`.

**Comment by jabbera** [comment]
@minrk I tried to grok the self.url change but the extra_path on wait_up seemed less risky and more generally useful as an api feature.

File: jupyterhub/objects.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # Distributed under the terms of the Modified BSD License.
 import socket
 import warnings
-from urllib.parse import urlparse, urlunparse
+from urllib.parse import urljoin, urlparse, urlunparse
```
Comment (minrk) [inline jupyterhub/objects.py:6]: Let's use `jupyterhub.utils.url_path_join` here, which I think is more likely to have the desired effect. `urljoin` behaves like relative URL resolution on a webpage, whereas `url_path_join` does a simpler path-style join.

For example:

```python
In [9]: urljoin("a/b", "c")
Out[9]: 'a/c'

In [10]: url_path_join("a/b", "c")
Out[10]: 'a/b/c'
```

**Comment by minrk** [comment]
extra_path makes sense! One minor comment about what join function to use, but 👍 after that.

**Pull request merged**